syncapi/consumers/roomserver.go

```diff
@@ -81,11 +81,18 @@ func (s *OutputRoomEventConsumer) onMessage(msg *sarama.ConsumerMessage) error {
 
 	switch output.Type {
 	case api.OutputTypeNewRoomEvent:
+		// Ignore redaction events. We will add them to the database when they are
+		// validated (when we receive OutputTypeRedactedEvent)
+		if output.NewRoomEvent.Event.Type() == gomatrixserverlib.MRoomRedaction && output.NewRoomEvent.Event.StateKey() == nil {
```

Comment:
Is `output.NewRoomEvent.Event.StateKey() == nil` right here or can it also be `""` as well as `nil`?

Reply:
It _has_ to be nil. Redactions are not state events so if we get one we should not handle it as one.